change-validation/src/main/scala/com/normation/plugins/changevalidation/api/ChangeRequestApi.scala

```diff
@@ -93,87 +96,94 @@ object ChangeRequestApi       extends ApiModuleProvider[ChangeRequestApi] {
+      AuthorizationType.Deployer.Edit,
+      AuthorizationType.Validator.Write,
+      AuthorizationType.Validator.Edit
+    )
+    override def dataContainer: Option[String]          = Some("changeRequests")
+    override def name:          String                  = "declineChangeRequest"
   }
   final case object AcceptRequestsDetails  extends ChangeRequestApi with OneParam with StartsAtVersion3 with SortIndex  {
     val z              = implicitly[Line].value
     val description    = "Accept given change request"
     val (action, path) = POST / "changeRequests" / "{id}" / "accept"
 
-    override def authz: List[AuthorizationType] = List(AuthorizationType.Deployer.Write, AuthorizationType.Deployer.Edit, AuthorizationType.Validator.Write, AuthorizationType.Validator.Edit)
-    override def dataContainer: Option[String] = None
+    override def authz:         List[AuthorizationType] = List(
+      AuthorizationType.Deployer.Write,
+      AuthorizationType.Deployer.Edit,
+      AuthorizationType.Validator.Write,
+      AuthorizationType.Validator.Edit
+    )
+    override def dataContainer: Option[String]          = Some("changeRequests")
+    override def name:          String                  = "acceptChangeRequest"
   }
   final case object UpdateRequestsDetails  extends ChangeRequestApi with OneParam with StartsAtVersion3 with SortIndex  {
     val z              = implicitly[Line].value
     val description    = "Update information about given change request"
     val (action, path) = POST / "changeRequests" / "{id}"
 
-    override def authz: List[AuthorizationType] = List(AuthorizationType.Deployer.Write, AuthorizationType.Deployer.Edit, AuthorizationType.Validator.Write, AuthorizationType.Validator.Edit)
-    override def dataContainer: Option[String] = None
+    override def authz:         List[AuthorizationType] = List(
+      AuthorizationType.Deployer.Write,
+      AuthorizationType.Deployer.Edit,
+      AuthorizationType.Validator.Write,
+      AuthorizationType.Validator.Edit
+    )
+    override def dataContainer: Option[String]          = Some("changeRequests")
+    override def name:          String                  = "updateChangeRequest"
   }
 
   def endpoints = ca.mrvisser.sealerate.values[ChangeRequestApi].toList.sortBy(_.z)
 }
 
 class ChangeRequestApiImpl(
-    restExtractorService: RestExtractorService,
+    diffService:          DiffService,
+    readTechnique:        TechniqueRepository,
     readChangeRequest:    RoChangeRequestRepository,
     writeChangeRequest:   WoChangeRequestRepository,
     readWorkflow:         RoWorkflowRepository,
-    writeWorkflow:        WoWorkflowRepository,
-    readTechnique:        TechniqueRepository,
     workflowLevelService: WorkflowLevelService,
     commitRepository:     CommitAndDeployChangeRequestService,
-    restDataSerializer:   RestDataSerializer
+    userPropertyService:  UserPropertyService,
+    currentUser:          CurrentUser
```

Comment:
you should not pass `CurrentUser` in a constructor. `CurrentUser` is an entity (a strange one, with a very strange persistance and retrieval model) but it's still an entity. 
Only parameters or other services are passed to service constructor. If you want to pass something to get current user (which is ok, it's not great to directly refer a magic entity in code), you should pass `com.normation.rudder.UserService`